### DISCUSSION TITLE
How to use custom column filter function with columnFilterModeOptions

### DISCUSSION THREAD

**Discussion by SRGajjala** [KevinVandy/material-react-table#31] [General]
when we set columnFilterModeOptions to  ['between', 'greaterThan', 'lessThan'] and filterFn:between with serverside paging and filtering, I am able to get the request parameter filters set to {id:"x",value:{[5,10]} and no idea what filtermode is set if mode is changed on client side.

If we can trigger customfunction with columnFilterModeOptions being set, I can pass one more key for filter condition in {id:"x",value:{[5,10],condition:'between'}

and there is one issue with  between filter mode, when we change anything in max textbox is not triggering filterchange event.

**Comment by KevinVandy** [comment]
This is actually one of the last big filter features I've yet to finalize. The ability to customize the filter modes available in the list is going to improve greatly soon.

As far as being able to tell which filter mode is chosen, you can do that by managing the `currentFilterFns` state yourself. More info will be available in [these docs](https://www.material-react-table.com/docs/guides/table-state-management) soon on how to do that.

I'll look into that last issue you described

**Reply by SRGajjala** [reply]
can you please notify me when the last issue is fixed

**Reply by KevinVandy** [reply]
I cannot replicate the last issue actually. Do you have an example of it?

**Reply by SRGajjala** [reply]
suppose we have column defined with **server side paging and filtering**       {
            id: 'age',
            header: 'age',
            columnFilterModeOptions: ['between', 'greaterThan', 'lessThan'],
          filterFn: 'between' 
}

this shows two **text fields** **min** and **max**. while changing value in **min** text field is triggering **API call,** but not with the change of value of **max** text field. please check change of value with min text field trigger API call always.

I followed remote data example https://www.material-react-table.com/docs/examples/remote
